src/rgw/rgw_op.cc

```diff
@@ -2524,6 +2524,17 @@ void RGWCreateBucket::execute()
     return;
   }
 
+  if(!placement_rule.empty()) {
+    if (!store->get_zonegroup().has_placement_target(placement_rule)) {
```

Comment:
could avoid the new `has_placement_target()` and reduce a level of nesting with:
```c++
  const auto& zonegroup = store->get_zonegroup();
  if (!placement_rule.empty() &&
      !zonegroup.placement_targets.count(placement_rule)) {
```